src/renderer/components/Scrubber.tsx

```diff
@@ -0,0 +1,139 @@
+import { useMemo, useState } from 'react';
+import SliderUnstyled, {
+  sliderUnstyledClasses,
+} from '@mui/base/SliderUnstyled';
+import { styled } from '@mui/material';
+import { secondToTimestampUI } from 'main/timeUtils';
+import colors from 'renderer/colors';
+import { roundToMs } from 'sharedUtils';
+
+const Slider = styled(SliderUnstyled)({
+  width: '100%',
+  padding: '10px 0',
+  display: 'inline-block',
+  position: 'relative',
+  cursor: 'pointer',
+
+  [`&:hover .${sliderUnstyledClasses.thumb}`]: {
+    opacity: '1',
+    transition: 'opacity 0.2s ease',
+  },
+
+  [`& .${sliderUnstyledClasses.rail}`]: {
+    display: 'block',
+    position: 'absolute',
+    width: '100%',
+    height: '5px',
+    backgroundColor: colors.grey[300],
+  },
+
+  [`& .${sliderUnstyledClasses.track}`]: {
+    display: 'block',
+    position: 'absolute',
+    height: '5px',
+    backgroundColor: `${colors.yellow[500]}CC`,
+  },
+
+  [`& .${sliderUnstyledClasses.thumb}`]: {
+    position: 'absolute',
+    width: '15px',
+    height: '15px',
+    marginLeft: '-5px',
+    marginTop: '-5px',
+    borderRadius: '50%',
+    backgroundColor: colors.yellow[500],
+    opacity: '0',
+    transition: 'opacity 0.2s ease',
+  },
+});
+
+const TimestampContainer = styled('div')({
+  display: 'flex',
+  flex: 1,
+  alignItems: 'flex-start',
+  justifyContent: 'space-between',
+  marginTop: -5,
+});
+
+const Timestamp = styled('div')({
+  fontFamily: '"Roboto Mono", monospace',
+  fontSize: 12,
+});
+
+interface ScrubberProps {
+  totalDuration: number;
+  currentTimeSeconds: number;
+  onScrubberChange: (newTime: number) => void;
+}
+
+const Scrubber = ({
+  totalDuration,
+  currentTimeSeconds,
+  onScrubberChange,
+}: ScrubberProps) => {
+  const [showRemainingTime, setShowRemainingTime] = useState(false);
+
+  const sliderValue = useMemo(() => {
+    // Prevent the slider from going past the end of the video
+    const currentTime =
+      currentTimeSeconds > totalDuration ? totalDuration : currentTimeSeconds;
+
+    // Round to two dp to avoid precision errors
+    const percentageCompleted =
```

Comment:
Make a util for this. Also, what's epsilon needed for?

Reply:
Moved into util and the epsilon has been removed.